src/Aspire.Hosting.MongoDB/MongoDBServerResource.cs

```diff
@@ -21,9 +21,61 @@ public class MongoDBServerResource(string name) : ContainerResource(name), IReso
     /// <summary>
     /// Gets the connection string for the MongoDB server.
     /// </summary>
-    public ReferenceExpression ConnectionStringExpression =>
-        ReferenceExpression.Create(
-            $"mongodb://{PrimaryEndpoint.Property(EndpointProperty.Host)}:{PrimaryEndpoint.Property(EndpointProperty.Port)}");
+    public ReferenceExpression ConnectionStringExpression
+    {
+        get
+        {
+            const string MongoScheme = "mongodb://";
+
+            var builder = new ReferenceExpression.ExpressionInterpolatedStringHandler(MongoScheme.Length, 2);
+
+            AppendConnectionString(builder);
+            AppendSuffix(builder);
+
+            return ReferenceExpression.Create(builder);
+        }
+    }
+
+    internal void AppendConnectionString(in ReferenceExpression.ExpressionInterpolatedStringHandler builder)
+    {
+        builder.AppendLiteral("mongodb://");
```

Comment:
There's a type called [ReferenceExpressionBuilder](https://learn.microsoft.com/en-us/dotnet/api/aspire.hosting.applicationmodel.referenceexpressionbuilder?view=dotnet-aspire-8.0) that you can use instead.

Reply:
oh much cleaner. thanks